services:
  db:
    image: postgres:latest
    restart: unless-stopped
    volumes:
      - postgres-data:/var/lib/postgresql/data
    environment:
      POSTGRES_USER: postgres
      POSTGRES_DB: postgres
      POSTGRES_PASSWORD: postgres
    ports:
      - "5432:5432"
    networks:
      - dev-network

  redis:
    image: redis:latest
    ports:
      - "6379:6379"
    networks:
      - dev-network
    healthcheck:
      test: ["CMD", "redis-cli", "ping"]
      interval: 5s
      timeout: 5s
      retries: 5

  celery:
    build:
      context: .
    depends_on:
      redis:
        condition: service_healthy
      db:
        condition: service_started
    command: celery -A core worker --loglevel=info
    environment:
      - DJANGO_SETTINGS_MODULE=core.settings.docker_dev
      - CELERY_BROKER_URL=redis://redis:6379/0
      - CELERY_RESULT_BACKEND=redis://redis:6379/0
      - PYTHONPATH=/workspaces/USE_CASE_COLLECTION
    env_file:
      - .env
    networks:
      - dev-network
    volumes:
      - .:/app

networks:
  dev-network:
    driver: bridge

volumes:
  postgres-data:
